compiler/modules/types/core.bal

```diff
@@ -815,13 +815,66 @@ function comparableNillableList(Context cx, SemType t1, SemType t2) returns bool
     }
     ComparableMemo memo = { semType1: t1, semType2: t2 };
     cx.comparableMemo.add(memo);
-    // SUBSET need to iterate members when tuples are supported
-    // following relies on the fact `listMemberType(cx, NIL, ()) = NEVER`
-    boolean result = comparable(cx, listMemberType(cx, t1, INT), listMemberType(cx, t2, INT));
+    boolean result;
+    if listAtomicTypeRw(cx, t1) == () || listAtomicTypeRw(cx, t2) == () {
```

Comment:
Why are we using listAtomicTypeRw here? That is tested in listAllMemberTypes.

Reply:
I am using it as a way to detect if the list type is not atomic (If this is not nil current placeholder function should work). When the proper version of `listAllMemberTypes` is implemented this whole if else part can be replaced with the body of else